Add Distributed Workload Metrics Ui Tests

Srihari1192 opened the pull request:
closes [RHOAIENG-4837](https://issues.redhat.com/browse/RHOAIENG-4837)

- Added Distributed workload Metrics basic tests to check with **admin and non admin users**
-  Added common methods and required xpath variables in **WorkloadMetricsUi.resource** page

Srihari1192 commented on ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:
```diff
@@ -0,0 +1,111 @@
+    Set Library Search Order    SeleniumLibrary
+    RHOSi Setup
+    ${rc}    ${out}=    Run And Return Rc And Output
+    ...     oc patch OdhDashboardConfig odh-dashboard-config -n ${APPLICATIONS_NAMESPACE} --type=merge -p '{"spec": {"dashboardConfig": {"disableDistributedWorkloads": false}}}'
```
This can be removed later once disableDistributedWorkloads enabled by default.

Srihari1192 commented on ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:
```diff
@@ -0,0 +1,111 @@
+Project Suite Teardown
+    [Documentation]    Suite teardown steps after testing Distributed Workload metrics .
+    ${rc}    ${out}=    Run And Return Rc And Output
+    ...     oc patch OdhDashboardConfig odh-dashboard-config -n ${APPLICATIONS_NAMESPACE} --type=merge -p '{"spec": {"dashboardConfig": {"disableDistributedWorkloads": true}}}'
```
This can be removed later once disableDistributedWorkloads enabled by default.

FedeAlonso commented on ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:
```diff
@@ -0,0 +1,115 @@
+Verify Workload Metrics Home page Contents
+    [Documentation]    Verifies "Workload Metrics" page is accessible from
+    ...                the navigation menu on the left and page contents
+    [Tags]    Sanity
```
Could you add some tag in the tests? RHOAIENG-4837 could be the good one (like it's done [here](https://github.com/red-hat-data-services/ods-ci/blob/4fee8773623707d9f1c6fecf291fd6663d960b1e/ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot#L34))

Srihari1192 replied:
sure will update the Tags , Thanks

Srihari1192 replied:
@FedeAlonso Can you review this Updated Tags

sutaakar commented on ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:
```diff
@@ -0,0 +1,115 @@
+    Select Distributed Workload Project By Name    ${PRJ_TITLE_NONADMIN}
+    Check Project Metrics Default Page Contents
+    Check Distributed Workload Status Page Contents
+    [Teardown]    Run Keywords
+    ...    Delete Data Science Project   ${PRJ_TITLE_NONADMIN}
+    ...    AND
+    ...    Wait Until Data Science Project Is Deleted  ${PRJ_TITLE_NONADMIN}
+    ...    AND
+    ...    Switch Browser    1
```
Shouldn't this be part of `Project Suite Teardown` rather than being part of the test itself?

Srihari1192 replied:
We can delete all the created Projects in `Project Suite Teardown `using method `Delete Data Science Projects From CLI` but thinking  if we ran tests independently this project may not be available and test fails

sutaakar replied:
ok

mturley commented on ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:
```diff
@@ -0,0 +1,115 @@
+    FOR  ${status}  IN   @{STATUS_LIST}
+        Page Should Contain    text=${status}: 0
+    END
+    Page Should Contain    No distributed workloads match your filters
```
This text may change before release, will let you know. We don't have filters implemented yet in the table so we may be replacing the whole table with an empty state message when there are no workloads.

mturley replied:
FYI @Srihari1192 now that https://github.com/opendatahub-io/odh-dashboard/pull/2686 is merged, this text is now simply "No distributed workloads"

sutaakar commented on ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:
```diff
@@ -0,0 +1,115 @@
+
+Verify That Not Admin Users Can Access Distributed workload metrics default page contents
+    [Documentation]    Verify That Not Admin Users Can Access Distributed workload metrics default page contents
+    [Tags]    Sanity
```
I think that this may be executed as part of Tier 1, no need to run as sanity test.
This test isn't a core basic test.

Srihari1192 replied:
Yes will update the Tags

sutaakar commented on ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:
```diff
@@ -0,0 +1,117 @@
+Project Suite Setup
+    [Documentation]    Suite setup steps for testing Distributed workload Metrics UI
+    Set Library Search Order    SeleniumLibrary
+    Skip If Component Is Not Enabled    kueue
```
Does it mean that whole test suite is skipped or just setup?

Srihari1192 replied:
Whole test suite is skipped.. I added this because DW Ui is disabled if kueue is `Removed`

Srihari1192 commented on ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:
```diff
@@ -0,0 +1,117 @@
+    ${rc}    ${out}=    Run And Return Rc And Output
+    ...     oc patch OdhDashboardConfig odh-dashboard-config -n ${APPLICATIONS_NAMESPACE} --type=merge -p '{"spec": {"dashboardConfig": {"disableDistributedWorkloads": false}}}'
+    Should Be Equal As Integers    ${rc}         ${0}
+    Sleep  1min  reason=Wait For Distributed workload metrics page to enable
```
yes ideally it update existing dashboard pods and already in running state so adding condition to pod state also not the right approach i think.. 
Default enabling DW UI will be available with this  [PR](https://github.com/opendatahub-io/odh-dashboard/pull/2685) merge soon

sutaakar replied:
So, once the linked PR is merged then there won't be a need for any sleep step, right?

Srihari1192 replied:
Yes we can remove this sleep once its merged

sutaakar replied:
Looks like that PR should be hopefully merged soon.
Do you want to wait for it to merge first or rather merge this PR now and remove sleep as follow up?

Srihari1192 replied:
Yes we can merge this PR for now .. i will update later once both changes [pr1](https://github.com/opendatahub-io/odh-dashboard/pull/2685) and [pr2](https://github.com/opendatahub-io/odh-dashboard/pull/2686) available in rhoai builds

mturley replied:
https://github.com/opendatahub-io/odh-dashboard/pull/2685 is now merged, builds will have the feature flag on by default going forward

Srihari1192 commented:
Added Kueue resources setup from the [PR](https://github.com/project-codeflare/codeflare-sdk/pull/505) Thanks @ChughShilpa for implementing this.
Updated the PR according to the new UI changes https://github.com/opendatahub-io/odh-dashboard/pull/2686 and https://github.com/opendatahub-io/odh-dashboard/pull/2685

sutaakar commented on ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:
```diff
@@ -0,0 +1,152 @@
+    Create Data Science Project    title=${PRJ_TITLE}    description=${PRJ_DESCRIPTION}
+    Set Global Variable    ${project_created}    True
+    # setup Kueue resource for the created project
+    Setup Kueue Resources    ${PRJ_TITLE}
```
Can you make sure that resources are also deleted after test execution?

Srihari1192 replied:
sure

Srihari1192 replied:
tested in PR check CI pipeline `rhods-ci-pr-test/2703`

sutaakar commented on ods_ci/tests/Resources/Page/DistributedWorkloads/kueue_resources_setup.sh:
```diff
@@ -0,0 +1,55 @@
+      resourceGroups:
+      - coveredResources: ["cpu", "memory", "nvidia.com/gpu"]
+        flavors:
+        - name: "default-flavor-mnist"
```
```suggestion
        - name: $flavor
```

Srihari1192 replied:
good catch , Thanks

sutaakar commented:
One fix above, otherwise looks good

ChughShilpa reviewed (approved):
Tested locally  and passed all tests
